test/check-application

```diff
@@ -27,37 +27,48 @@ class TestApplication(testlib.MachineCase):
 
         # show image listing toggle
         b.wait_present('#containers-images tr:contains("busybox:latest")')
-        b.wait_visible('#containers-images tr:contains("busybox:latest")')
         b.click('#containers-images tbody tr:contains("busybox:latest") td.listing-ct-toggle')
         b.wait_present('#containers-images tbody tr:contains("busybox:latest") + tr button.btn-delete')
 
         # run a container (will exit immediately)
-        m.execute("podman run -d --name swamped-crate busybox sh")
+        m.execute("podman run -d --name test-sh alpine sh")
+        # run a container
+        m.execute("podman run -d --name swamped-crate busybox sleep 1000")
 
         # FIXME: UI needs to listen to change signals and update automatically
+
         b.reload()
-        b.wait_present("#content")
         b.enter_page("/podman")
         b.wait_present("#containers-containers-filter button")
 
-        # show all containers to interact with stopped ones
+        # show all containers and check status
         b.click("#containers-containers-filter button")
-        b.wait_visible("#containers-containers-filter .dropdown-menu")
+        b.wait_present("#containers-containers-filter .dropdown-menu")
         b.click("#containers-containers-filter li[data-data='all'] a")
         b.wait_present("#containers-containers")
+        # check containers name
+        b.wait_present('#containers-containers tr:contains("swamped-crate")')
         b.wait_present('#containers-containers tr:contains("busybox:latest")')
-        b.wait_visible('#containers-containers tr:contains("busybox:latest")')
-
+        b.wait_present('#containers-containers tr:contains("test-sh")')
+        b.wait_present('#containers-containers tr:contains("alpine:latest")')
+        # check busybox is running and alpine is exited
+        b.wait_present('#containers-containers tr:contains("busybox:latest") td:contains("running")')
+        b.wait_present('#containers-containers tr:contains("alpine:latest") td:contains("exited")')
+        # check container command
+        b.wait_present('#containers-containers tr:contains("busybox:latest") td:contains("sleep 1000")')
+        b.wait_present('#containers-containers tr:contains("alpine:latest") td:contains("sh")')
         # show container listing toggle
         b.click('#containers-containers tbody tr:contains("busybox:latest") td.listing-ct-toggle')
         b.wait_present('#containers-containers tbody tr:contains("busybox:latest") + tr button.btn-delete')
 
+        # show running container
         b.click("#containers-containers-filter button")
         b.wait_present("#containers-containers-filter .dropdown-menu")
         b.click("#containers-containers-filter li[data-data='running'] a")
-        b.wait_present("#containers-containers")
-        b.wait_in_text("#containers-containers", "No running containers")
```

Comment:
Please keep this case as well, it's important to ensure that the UI reacts properly to the "no containers" case. I suggest the following:

  1. Ensure that "No running containers" is shown.
  2. Start a running one, reload, ensure it's shown.
  3. Start the other one that immediately exits, reload, ensure it's *not* shown.
  4. Change the dropdown to "everything", ensure that the exited container is then shown and has expected properties.
  5. Change the dropdown back to "running only" and ensure that the latter goes away again.
  6. Stop the running container, reload, ensure that the running one goes away and "no running containers" is shown again.

I think this should cover all possible transitions. Feel free to introduce a helper like

    def check_container(row_name, expected_strings)

which you call like

     check_container("alpine-latest", ["running", "sleep 1000"])

this avoids some repetition.